a8c-files.php

```diff
@@ -1156,6 +1156,59 @@ function a8c_files_maybe_inject_image_sizes( $data, $attachment_id ) {
 
 	return $data;
 }
+/**
+ * Filter dimensions for images to return a proper image size and srcset values
+ *
+ * @param bool   $value         The filtered value, defaults to `true`.
+ * @param string $image         The HTML `img` tag where the attribute should be added.
+ * @param string $context       Additional context about how the function was called or where the img tag is.
+ * @param int    $attachment_id The image attachment ID.
+ */
+function a8c_files_maybe_inject_dimensions( $value, $image, $context, $attachment_id ) {
+	$vipgo_img_src = preg_match( '/src="([^"]+)"/', $image, $match_src ) ? $match_src[1] : '';
+
+	/** This filter is documented in wp-admin/includes/media.php */
+	add_filter( 'wp_image_src_get_dimensions', function( $dimensions, $image_src, $image_meta, $attachment_id ) use ( $vipgo_img_src ){
```

Comment:
This closure is like 90% of the parent function.

Consider extracting it a named function, so that it can then be properly tested.